### PR TITLE
fix(Android): jumping content on fabric

### PR DESCRIPTION
**Pull request by alduzy**
## Description

This PR applies modifications to a previous fix: https://github.com/software-mansion/react-native-screens/pull/2169 for fabric only, which has stopped working since RN `0.75`.

In RN `0.74` the `adopt` in `RNSScreenComponentDescriptior.h` was once called without `stateData` but with children and we could then check if the `ScreenStackHeaderConfig` is present among them and make adjustments based on it.

When working on https://github.com/software-mansion/react-native-screens/pull/2292 it became clear that the fix does not work anymore. Now the `adopt` is called either with no children and no `stateData` or with both.
The solution is to move the code to `appendChild` in `RNSScreenShadowNode.cpp` so we can perform the adjustments as soon as the children append.

## Changes

- moved code from `adopt` in `RNSScreenComponentDescriptior.h` to newly added `appendChild` override in `RNSScreenShadowNode.cpp`

### Before

https://github.com/user-attachments/assets/6b76864b-58bb-4c6e-9f5b-a01bb0c88d2a

### After

https://github.com/user-attachments/assets/98931e77-3877-4f67-8b28-f49d2e0f42ff

## Test code and steps to reproduce

- Use `TestHeader` to test this change

## Checklist

- [ ] Included code example that can be used to test this change
- [ ] Updated TS types
- [ ] Updated documentation: 
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/guides/GUIDE_FOR_LIBRARY_AUTHORS.md
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/native-stack/README.md
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/src/types.tsx
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/src/native-stack/types.tsx
- [ ] Ensured that CI passes

### CODE REVIEW

**Comment by tboba** [comment]
Why do we want to merge this into `@kkafar/fabric-pressables`? Is this PR some part of ongoing another one?

**Comment by WoLewicki** [comment]
Is there a reason you removed the checks for if the calculated values are the same as the ones returned by the system later? It is important source of information for if we do the calculations right.

**Comment by alduzy** [comment]
@tboba You're right, we can merge it independently.

File: common/cpp/react/renderer/components/rnscreens/RNSScreenShadowNode.cpp [hunk 1]
```diff
@@ -15,6 +15,110 @@ Point RNSScreenShadowNode::getContentOriginOffset(
...
+                        << "[RNScreens] Failed to retrieve packageInstance or the package instance was null on JVM side";
+                return {};
+            }
+
+            jfloat headerHeight = env->CallFloatMethod(
+                    packageInstance, computeDummyLayoutID, fontSize, isTitleEmpty);
+
+            return {headerHeight};
+        }
+#endif // ANDROID
```
Comment (tboba) [inline common/cpp/react/renderer/components/rnscreens/RNSScreenShadowNode.cpp:85]: Let's move these methods below `appendChild`

**Comment by tboba** [comment]
Are we sure this is correct solution? Currently, my emulator on `TestHeader` hides header behind the status bar. Is it just my case, or this correction should be reconsidered?

CleanShot 2024-08-23 at 11 48 08@2x

**Review by tboba — APPROVED** [review]
LGTM! Just one remark about method ordering 😄

File: common/cpp/react/renderer/components/rnscreens/RNSScreenShadowNode.cpp (hunk 1 above)
Comment (alduzy) [reply common/cpp/react/renderer/components/rnscreens/RNSScreenShadowNode.cpp:85]: @tboba I'm afraid those methods have to be declared before they're used in cpp

**Comment by alduzy** [comment]
@tboba EDIT: the correct layout is actually achieved based on the changes made in https://github.com/software-mansion/react-native-screens/pull/2292 so I had to rebase back to it.

File: common/cpp/react/renderer/components/rnscreens/RNSScreenShadowNode.cpp (hunk 1 above)
Comment (tboba) [reply common/cpp/react/renderer/components/rnscreens/RNSScreenShadowNode.cpp:85]: @alduzy But didn't we achieve different order, but in RNSScreenComponentDescriptor.h? 😄 
It's not a big deal to have them here though, so if it's really not possible, I think we can proceed with what we have right now.

**Review by kkafar — APPROVED** [review]
Looks good! Thanks!

**Pull request merged**